build-config/resources/Ballerina.toml

```diff
@@ -21,3 +21,6 @@ path = "../native/build/libs/salesforce-native-@project.version@.jar"
 groupId = "io.ballerinax"
 artifactId = "salesforce"
 version = "@project.version@"
+
+[[platform.java17.dependency]]
+path = "./lib/opencsv-5.9.jar"
```

Comment:
IIRC Its recommended to define the native dependencies including the group ID and artifact ID separately (e.g. https://github.com/ballerina-platform/module-ballerinax-redis/blob/master/build-config/resources/Ballerina.toml)